src/NuGet.Core/NuGet.Frameworks/FrameworkReducer.cs

```diff
@@ -389,18 +396,21 @@ private IEnumerable<NuGetFramework> GetNearestPCLtoPCL(NuGetFramework framework,
                 // from all possible frameworks, find the best match
                 var nearestForSub = GetNearest(sub, allPclFrameworks);
 
-                // +1 each framework containing the best match
-                foreach (var pair in pclToFrameworks)
+                if (nearestForSub != null)
```

Comment:
I don't think it's possible for `nearestForSub` to be null here, but that's because in the call to `GetNearest` the arguments will always have a result, whereas with different inputs `GetNearest()` could return null, hence it's annotated in that way. Anyway, in order to keep the compiler happy, something had to be done, but I think that `paris.Value.Contains` would have thrown an `ArgumentException`.

It only occurs to me now, that I could have used `!` to suppress the analyzer and keep old behaviour. Does anyone think I should do that instead?  Or throw an exception with a more clear error message?  If noone replies to this comment, I'll assume people are fine with this change.

Reply:
This is probably better, I don't think the rest of the code would have thrown if `nearestForSub` is `null` since it only passes it to `Contains` which just returns `false`.  At least now the lookup is skipped.

Reply:
Oops, you're right. `pairs.Value` is an `IEnumerable`, not a `Dictionary`. `Dictionary.ContainsKey` throws on null, but that's not happening here.